### DISCUSSION TITLE
Lua command to update all installed packages?

### DISCUSSION THREAD

**Discussion by ghost** [mason-org/mason.nvim#909] [Q&A] [answered]
I'd like to update all installed Mason packages whenever I `PackerUpdate`, rather than having to open the Mason UI and press `U`. What do I put in the `run` field of the package netry to accomplish this?

**Comment by williamboman** [comment, marked as answer]
Hey! There exists no such command yet. You could either write your own, or use something like https://github.com/WhoIsSethDaniel/mason-tool-installer.nvim or https://github.com/RubixDev/mason-update-all

**Reply by ghost** [reply]
Oh, unfortunate. Thanks for the tips though. Is there a plan to add such a command?

**Reply by williamboman** [reply]
Yeah eventually. There are some things that I think need to be in place before people start upgrading packages so carelessly <sup>(I don't mean that in a bad way at all, more in the literal sense)</sup>. In the meantime, if this is something you really want I'd recommend looking into coding it together yourself in your config. [The Lua APIs in Mason](https://github.com/williamboman/mason.nvim/blob/main/doc/reference.md) are very high-level, and it shouldn't require more than 10-15 lines, depending on what you want to do of course.

**Reply by ghost** [reply]
Ok, what do you think of this: https://github.com/EleanorNB/nvim/blob/trunk/lua/languages.lua#L40-L56

**Reply by williamboman** [reply]
Looks good to me!
